manifest/oci.go

```diff
@@ -259,9 +260,44 @@ func (m *OCI1) ImageID([]digest.Digest) (string, error) {
 	if err := m.Config.Digest.Validate(); err != nil {
 		return "", err
 	}
+
+	// If there is any layer that is using partial content, we calculate the image ID
+	// in a different way since the diffID cannot be validated as for regular pulled images.
+	for _, layer := range m.Layers {
```

Comment:
I think this is not quite correct; it doesn’t matter that the layer _has_ a TOC, it matters whether we have _used and verified_ the TOC vs. the uncompressed digest.

So, the same manifest might turn into several different image IDs depending on whether partial pulls succeeded; and that per-pull-per-layer information will need to be maintained by `storageImageDestination`.

Reply:
This was resolved in #2288.